src/turbomind/models/llama/LlamaWeight.cc

```diff
@@ -82,6 +82,15 @@ LlamaWeight<T>::~LlamaWeight()
     decoder_layer_weights.clear();
 
     // Wait for deallocations
+    check_cuda_error(cudaStreamSynchronize(stream_));
+
+    // Trim mempool
+    int           device_id;
+    cudaMemPool_t mempool{};
+    check_cuda_error(cudaGetDevice(&device_id));
```

Comment:
Should we apply device guard here?

Reply:
already has https://github.com/InternLM/lmdeploy/blob/de74d209c7e6e167054afd051c13d94e80de96a4/src/turbomind/triton_backend/llama/LlamaTritonModel.cc#L166-L168